commands/invoke_test.go

```diff
@@ -25,26 +25,58 @@ func Test_invoke(t *testing.T) {
 			ResponseStatusCode: http.StatusOK,
 			ResponseBody:       expected_invoke_response,
 		},
+		{
+			Method:             http.MethodPost,
+			Uri:                "/async-function/" + funcName,
+			ResponseStatusCode: http.StatusOK,
+			ResponseBody:       expected_invoke_response,
+		},
 	})
 	defer s.Close()
 
-	os.Stdin, _ = ioutil.TempFile("", "stdin")
-	os.Stdin.WriteString("test-data")
-	os.Stdin.Seek(0, 0)
-	defer func() {
-		os.Remove(os.Stdin.Name())
-	}()
-
-	stdOut := test.CaptureStdout(func() {
-		faasCmd.SetArgs([]string{
-			"invoke",
-			"--gateway=" + s.URL,
-			funcName,
+	t.Run("Sync", func(t *testing.T) {
```

Comment:
Rather than `t.Run` in this instance for consistency I'd prefer two separate test methods. Please could you consider changing this?

Reply:
As above, have reverted to the old style